### PR TITLE
Fix bad box extents in KsgsM84Src

### PR DESCRIPTION
**Pull request by neilmatula**

### CODE REVIEW

**Review by tonyinme — APPROVED** [review]
Looks good. I had a minor comment that you can address if you want.

File: amr-wind/equation_systems/tke/source_terms/KsgsM84Src.cpp [hunk 1]
```diff
@@ -68,12 +69,17 @@ void KsgsM84Src::operator()(
...
         }
 
         amrex::Orientation ohi(dir, amrex::Orientation::high);
-        if (bctype[ohi] == BC::wall_model) {
-            amrex::Box bhi = amrex::adjCellHi(bx, dir, 0) & bx;
+        if (bctype[ohi] == BC::wall_model &&
+            bx.bigEnd(dir) == geom.Domain().bigEnd(dir)) {
+            amrex::Box bhi = amrex::bdryHi(bx, dir, 1);
+            amrex::Abort(
+                "fix me later: need an upper version of lower_boundary_faces");
```
[the remark this answers is not in the document]
Comment (psakievich) [reply amr-wind/equation_systems/tke/source_terms/KsgsM84Src.cpp:82]: Agreed.

**Comment by tonyinme** [comment]
This looks good, I think it's ready to merge @michaeljbrazell .

**Comment by michaeljbrazell** [comment]
So this bug slipped through because we didn't have regression test coverage. Should we add a case that uses TKE with wall model? Also, has anyone confirmed this is the correct implementation? Maybe there is already a case out there that we could shrink down for a regression test? @gantech  ?

**Comment by michaeljbrazell** [comment]
let's add a reg test in a separate PR

**Pull request merged**